consensus/src/consensus/mod.rs

```diff
@@ -504,6 +509,52 @@ impl ConsensusApi for Consensus {
         self.headers_store.get_timestamp(self.get_sink()).unwrap()
     }
 
+    fn get_current_block_color(&self, hash: Hash) -> Option<bool> {
```

Comment:
1. prune guard is needed in order to make sure data is consistent and does not get pruned during the loop
2. pre condition: check that hash exists (see other methods in this file)

```suggestion
    fn get_current_block_color(&self, hash: Hash) -> Option<bool> {
        let _guard = self.pruning_lock.blocking_read();
        self.validate_block_exists(hash)?;
```